### ISSUE TITLE
Ploting graphs in terminal

### ISSUE THREAD

**Issue by RustemB** [Qalculate/libqalculate#214]
- Is it (or will) possible to plot graphs inside terminal session?

- Will it be possible to plot graphs inside GUI version of qalculate?

**Comment by hanna-kn** [comment]
Yes, and yes. It does however require that [Gnuplot](http://www.gnuplot.info/) is installed (and found in path).

You can either use to the plot() function, or File->Plot Functions/Data (in the GUI).

plot function documentation:
> _Plot Functions and Vectors_
> 
> **plot(Expression or vector[, Minimum x value][, Maximum x value][, Number of samples / Step size][, X variable][, Use step size][, Persistent])**
> 
> Plots one or more expressions or vectors. Use a vector for the first argument to plot multiple series. Only the first argument is used for vector series. It is also possible to plot a matrix where each row is a pair of x and y values.
> 
> Example: plot([x^2, 2x, [0,1,4,8,16]], 0, 4).
> 
>  **Arguments.**
> 
> - Expression or vector: a free value
> - Minimum x value: a real number (optional, default: 0)
> - Maximum x value: a real number (optional, default: 10)
> - Number of samples / Step size: a free value (optional, default: 1001)
> - X variable: an unknown variable/symbol (optional, default: x)
> - Use step size: a boolean (0 or 1) (optional, default: 0)
> - Persistent: a boolean (0 or 1) (optional, default: 0)
> 
> **Requirement.**  "Minimum x value" < "Maximum x value"

**Comment by RustemB** [comment]
But it's opens me GUI gnuplot, and i want "ascii-ish" inside of terminal, 'cuz it looks weird for me
![image](https://user-images.githubusercontent.com/25725953/89169844-3b02b580-d598-11ea-9d9e-070b43259867.png)

**Comment by hanna-kn** [comment]
> But it's opens me GUI gnuplot, and i want "ascii-ish" inside of terminal, 'cuz it looks weird for me

That is currently not possible (and will likely never be).

**Comment by RustemB** [comment]
sadly :(

**Comment by i-ky** [comment]
Actually, gnuplot itself supports ASCII output using `set terminal dumb ...`:
> The **dumb** terminal driver plots into a text block using ascii characters. It has an optional size specification
and a trailing linefeed flag.

Maybe it is not that difficult to implement? And maybe it's not that useless if gnuplot has this feature?

**Comment by hanna-kn** [comment]
> Maybe it is not that difficult to implement?

`echo "set term dumb" >> ~/.gnuplot` should be enough.

> Maybe it is not that difficult to implement? And maybe it's not that useless if gnuplot has this feature?

To me, this output appears close to useless:
![Skärmbild från 2020-08-03 13-34-37](https://user-images.githubusercontent.com/19703699/89178551-23beca80-d58e-11ea-996b-bd1885aab6ed.png)

**Comment by danr** [comment]
There are projects that use Braille symbols to make plots in the terminal, eg https://github.com/sunetos/TextPlots.jl https://github.com/asciimoo/drawille :woman_shrugging:

**Comment by yump** [comment]
On that note, the upcoming gnuplot 5.5 (not yet packaged in my distro) has added the [block](http://www.gnuplot.info/docs/loc18288.html) terminal, which uses that trick.  There's also the [sixelgd](http://www.gnuplot.info/docs/loc21341.html) terminal, which has been around longer and is somewhat more capable but less compatible (doesn't work in tmux, for example).

With `set term sixelgd truecolor size 480,280 font "sans,8"` in ~/.gnuplot it looks like

![Screenshot_20220719_020324](https://user-images.githubusercontent.com/2214413/179687995-631f559e-4817-49a2-b0ce-ecd34fc86f46.png)
